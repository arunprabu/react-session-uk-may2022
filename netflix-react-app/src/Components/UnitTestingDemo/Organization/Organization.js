import React, { useState } from 'react';

const Organization = (props) => {

  const [ myCountry, setMyCountry] = useState('USA');

  const handleCountryName = (event) => {
    setMyCountry(event.target.value);
  }

  return (
    <div className='text-center'>
      <h3>Welcome to our Organization's Official Page!!!</h3>
      <h5 data-testid="companyNameEl">{props.companyName}</h5>
      <p data-testid="employeeCountEl" 
      style={{ padding: '10px', color: 'green', textAlign: 'center'}}>Employees: 400000+</p>
      <a href="/" role='button' className='btn btn-primary btn-lg'>JOIN NOW</a>
      <br/><br/>
      <input type='text' placeholder='Enter Country Name' value={myCountry} onChange={handleCountryName}/>

      <p data-testid="visitEl">Visit Cognizant {myCountry} Website</p>
    </div>
  )
}

export default Organization