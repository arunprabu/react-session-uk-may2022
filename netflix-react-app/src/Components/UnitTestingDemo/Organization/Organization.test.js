import { fireEvent, render, screen } from '@testing-library/react';
import Organization from './Organization';
import renderer from 'react-test-renderer';

describe('OrganizationComponent', () => {
 
  // Test Spec #1 
  it('should be available within the App', () => {
    expect(Organization).toBeDefined();
  });

  // Test Spec #2
  it(`has 'Welcome to our Organization's Official Page!'`, () => {
    render(<Organization />);
    const officialTextEl = screen.getByText(`Welcome to our Organization's Official Page!!!`);
    expect(officialTextEl).toBeInTheDocument();
  });

  // Test Spec #3 // styles
  it('has css style with text color green', () => {
    render(<Organization />);
    expect(screen.getByTestId('employeeCountEl')).toHaveStyle('color:green');
  });

  // Test Spec #4 // css class
  it('has button with css class btn', () => {
    render(<Organization />);
    expect(screen.getByRole('button')).toHaveClass('btn');
  });

  // Test Spec #5 // props 
  it('receives props companyName and displays in template', () => {
    render(<Organization companyName='Cognizant Technologies'/>);
    expect(screen.getByTestId('companyNameEl')).toHaveTextContent('Cognizant Technologies');
  });

  // Test Spec #6 input with placeholder 
  it('has an input with placeholder Enter Country Name', () => {
    render(<Organization />);
    expect(screen.getByPlaceholderText('Enter Country Name')).toBeTruthy();
  });

  // Test Spec #7 - checking if the input el is working with onChange event and states
  it('should update country onChange Event', () => {
    render(<Organization />);
    // find the input element to handle onChange 
    const countryInput = screen.getByPlaceholderText('Enter Country Name');
    // now checking whether onChange is working or not
    // mock fire change event with the mock value 
    fireEvent.change(countryInput, { target: { value: 'Canada'}} );
    // finally checking if the input element is showing the entered country in JSX
    expect(countryInput.value).toBe('Canada');
    expect(screen.getByTestId('visitEl')).toHaveTextContent(`Visit Cognizant Canada Website`);
  });
  
  // snapshot testing
  it('should have right company snapshot', () => {
    // to take a snapshot we need a tool - react-test-renderer (npm i react-test-renderer@17.0.2 -D)
    // taking a snapshot and converting it to JSON
    // [RECOMMENDED]: Take snapshot with necessary props
    const snapshotJSON = renderer.create(<Organization companyName='Cognizant Technologies'/>).toJSON();
    // lets assert with a matcher -- toMatchSnapshot() from jest 
    expect(snapshotJSON).toMatchSnapshot();
  });

  // AJAX call - mocking 
  
});