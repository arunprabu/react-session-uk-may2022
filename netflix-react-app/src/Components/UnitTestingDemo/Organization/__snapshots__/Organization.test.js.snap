// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationComponent should have right company snapshot 1`] = `
<div
  className="text-center"
>
  <h3>
    Welcome to our Organization's Official Page!!!
  </h3>
  <h5
    data-testid="companyNameEl"
  >
    Cognizant Technologies
  </h5>
  <p
    data-testid="employeeCountEl"
    style={
      Object {
        "color": "green",
        "padding": "10px",
        "textAlign": "center",
      }
    }
  >
    Employees: 400000+
  </p>
  <a
    className="btn btn-primary btn-lg"
    href="/"
    role="button"
  >
    JOIN NOW
  </a>
  <br />
  <br />
  <input
    onChange={[Function]}
    placeholder="Enter Country Name"
    type="text"
    value="USA"
  />
  <p
    data-testid="visitEl"
  >
    Visit Cognizant 
    USA
     Website
  </p>
</div>
`;
